### PR TITLE
Add option to disable elastic authentication

### PR DESCRIPTION
**Pull request by eufebius**
## What was changed
Add condition to init container which need to check elasticsearch 

## Why?
To enable work with ES without authentication ( aws es service )

## Checklist

1. Closes https://github.com/temporalio/helm-charts/issues/200

2. How was this tested:
- tried deploy with $.Values.elasticsearch.authentication.enabled.true
- tried deploy with $.Values.elasticsearch.authentication.enabled.false

3. Any docs updates needed?

### CODE REVIEW

**Comment by Whitespace** [comment]
@eufebius I think your commit is a different email than the one that your GH account uses, and you have to sign the CLA with the other email.

File: values.yaml [hunk 1]
```diff
@@ -291,6 +291,8 @@ elasticsearch:
   port: 9200
   version: "v7"
   logLevel: "error"
+  authentication:
+    enabled: true
```
Comment (underrun) [inline values.yaml:295]: flip option to `.authentication.disabled` instead - should be enabled by default and if we want to disable this set to true

**Comment by underrun** [comment]
Depending on the final scope of changes that fall under https://github.com/temporalio/proposals/issues/42 this may be handled differently in the future. Ideally the elastic search auth method shouldn't be something that needs to be known by the helm chart, but I'm leaving this open for now so we're tracking that if there are any connections made to ES from inline code in the helm charts we need to account for auth strategy.

**Comment by underrun** [comment]
After talking about this, i don't think we'll need to have connections info directly in kubernetes resources like this going forward. Our new approach will ensure that we can connect to es either with or without user/pass

**Pull request closed without merging**